### PR TITLE
CONSOLE-3252: Create a ConsoleNotification when cluster is performing an upgrade

### PR DESCRIPTION
**Pull request by jhadvig**
This story creates a ConsoleNotification if the cluster is under upgrade process. Once the upgrade is done we remove the notification.

Screen:
Screenshot 2022-10-26 at 16 18 23

Story: https://issues.redhat.com/browse/CONSOLE-3252

/assign @TheRealJon

### CODE REVIEW

**Review by TheRealJon — APPROVED** [review]
/lgtm

**Comment by jhadvig** [comment]
QE Approver: 
/assign @yapei
Docs Approver:
 /assign @opayne1 
PX Approver: 
/assign @RickJWagner

**Comment by RickJWagner** [comment]
Thank you for providing a link to the associated JIRA CONSOLE-3252 and the nice screen shot.  Very easy to review.

**Comment by yanpzhan** [comment]
@jhadvig I launched a cluster against the pr, and upgrade it to another version, I could see the notification, but the console operator would be in Degrade status always. Is it because the target version doesn't contain the code for the notification?
Here is console operator message: ConsoleNotificationSyncDegraded: consolenotifications.console.openshift.io is forbidden: User "system:serviceaccount:openshift-console-operator:console-operator" cannot create resource "consolenotifications" in API group "console.openshift.io" at the cluster scope
Screenshot from 2022-10-27 18-05-33

**Comment by jhadvig** [comment]
@yanpzhan so I've tried to reproduce the issue but without any luck. Dont see any errors in the console operator.

Screenshot 2022-10-27 at 16 19 15

Could you please try to test again and ping me when you will be testing it? THanks

**Comment by yanpzhan** [comment]
@jhadvig from you screenshot, the console-operator doesn't start upgrade, the error appears when the console operator tries to upgrade.

**Comment by yanpzhan** [comment]
During testing, the notification is shown as designed. And I will test against payload after the pr's merged, so that ensure the upgrade could success on both origin version and target version containing the code.

**Pull request merged**
